uiowa/uiowa
# Commencement Event Migration

Closes #7567
Closes #7627

# How to test

- Sync down the D7 Commencement site in DevDesktop
```
ddev blt ds --site=commencement.uiowa.edu && ddev drush @commencement.local en commencement_migrate -y && ddev drush @commencement.local uli admin/config/sitenow/sitenow-migrate && ddev drush @commencement.local sqlq "SELECT COUNT(*) FROM paragraphs_item WHERE type = 'uiowa_collection_item'"
```
- ^ Note the number that is printed out after the login link, as that is your baseline count for `uiowa_collection_item` paragraphs.
- Configure SiteNow Migrate following instructions for DevDesktop. 
  - Database: `commencement_uiowa_edu`.
  - User: `root`
  - Host: `host.docker.internal`
  - Port: `33067`
  - Base URL: `https://commencement.uiowa.edu`
```
ddev drush @commencement.local mim --all && ddev drush @commencement.local sqlq "SELECT COUNT(*) FROM paragraphs_item WHERE type = 'uiowa_collection_item'"
```
- The query above will give you the count of `uiowa_collection_items` after running the migration. We expect a number in the 70s or above.
- Spot check events
- Ensure correct order of events are being pulled over
- Example of event with livestream and ceremony information
- Try some redirects by visiting the current site, getting the URL for a ceremony and using it on the local site to ensure that you are redirected as expected.

## Rollback cleanup
- Final check for `uiowa_collection_item` paragraphs:
```
ddev drush @commencement.local mr --all && ddev drush @commencement.local sqlq "SELECT COUNT(*) FROM paragraphs_item WHERE type = 'uiowa_collection_item'"
```
- Number should match the baseline from earlier.

## Additional updates
- https://commencement.uiowa.ddev.site/node/add/event
- Confirm that Ceremony information field is a text area
- Confirm that Ceremony information details displays below that, is a paragraph widget, and doesn't pre-populate a collection item.
- https://commencement.uiowa.ddev.site/node/add/venue
- Confirm that a collection section is not pre-populated under Additional information
- Click "Add Collection Section"
- Confirm that a collection item is not pre-populated under Collections
---

### Additional helpful commands

- Rollback migration
```
ddev drush @commencement.local mr commencement_event
```
- Stop and reset migration
```
ddev drush @commencement.local mst commencement_event
ddev drush @commencement.local mrs commencement_event
```
- Re-import migration config
```
ddev drush @commencement.local cim -y &&  && ddev drush @commencement.local cr
```
